### PR TITLE
CONJ-525 Fix for having extra update count during batch queries, with…

### PR DESCRIPTION
**Pull request by yerenkow**
… -3 which is unexpected by external consumers

### CODE REVIEW

**Comment by yerenkow** [comment]
Travis was fine for this commit
https://travis-ci.org/yerenkow/mariadb-connector-j/builds/275886739

I have no idea why appveyor has failed :)

**Comment by rusher** [comment]
Hi @yerenkow,

Correction will be in next version.

Similar to other open source projects, the MariaDB Foundation needs to have shared ownership of all code that is included in the MariaDB distribution. The easiest way to achieve this is by submitting your code under the BSD-new license. (The other alternative is to sign the code contribution agreement which can be found here: https://mariadb.com/kb/en/mariadb/mca/)

Please indicate in a comment below that you are contributing your new code of the whole pull request under the BSD-new license or that you have filled out the contribution agreement and sent it.

Thanks,
diego

File: src/main/java/org/mariadb/jdbc/internal/com/read/dao/CmdInformationBatch.java [hunk 1]
```diff
@@ -95,6 +95,16 @@ public void addErrorStat() {
         this.updateCounts.add((long) Statement.EXECUTE_FAILED);
     }
 
+    /**
+     * Clear error state, used for clear exception after first batch query, when fall back to per-query execution.
+     *
+     */
+    @Override
+    public void clearErrorStat() {
+        hasException = true;
```
Comment (rusher) [inline src/main/java/org/mariadb/jdbc/internal/com/read/dao/CmdInformationBatch.java]: reverting must then set hasException to false

**Comment by yerenkow** [comment]
I'm contributing new code and changes of the whole pull request under the BSD-new license.
Feel free to use it or rewrite at your convenience.

Thanks.

File: src/main/java/org/mariadb/jdbc/internal/com/read/dao/CmdInformationBatch.java (hunk 1 above)
Comment (yerenkow) [reply src/main/java/org/mariadb/jdbc/internal/com/read/dao/CmdInformationBatch.java]: My mistake, of course this should be set to false!

**Pull request merged**